### DISCUSSION TITLE
how to handle unmet typescript version peer dependency for angular 11 project?

### DISCUSSION THREAD

**Discussion by japj** [renovatebot/renovate#10437] [Request Help]
Hello,

I am using:
- self-hosted renovate (v25.41.2) with AzureDevOps Server 2020
- renovate.json containing:
```json
  "extends" : [
    "config:base",
    "group:allNonMajor"
  ]
```
- angular 11 project
- typescript 4.1.3 as a devDependency
- npm package-lock files
- nodejs 14.16.1 with npm 6.14.12

When renovate creates a new PR for "chore(deps): update all non-major dependencies" it seems it is also updating typescript to 4.3.2, this causes the build to fail since angular 11 requires TypeScript >=4.0.0 and <4.2.0 but 4.3.2 was found instead.

I see from the renovate logfile that `npm install --package-lock-only --no-audit --ignore-scripts` is run and that stderr has some relevant peer dependency information from npm:
```sh
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@~2.3.1 (node_modules\\rollup\\node_modules\\fsevents):\nnpm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@2.3.2: wanted {\"os\":\"darwin\",\"arch\":\"any\"} (current: {\"os\":\"win32\",\"arch\":\"x64\"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@^1.2.7 (node_modules\\watchpack-chokidar2\\node_modules\\chokidar\\node_modules\\fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.13: wanted {\"os\":\"darwin\",\"arch\":\"any\"} (current: {\"os\":\"win32\",\"arch\":\"x64\"})\nnpm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@^1.2.7 (node_modules\\webpack-dev-server\\node_modules\\chokidar\\node_modules\\fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.13: wanted {\"os\":\"darwin\",\"arch\":\"any\"} (current: {\"os\":\"win32\",\"arch\":\"x64\"})
npm WARN @angular-devkit/build-angular@0.1102.14 requires a peer of typescript@~4.0.0 || ~4.1.0 but none is installed. You must install peer dependencies yourself.
npm WARN @angular/compiler-cli@11.2.14 requires a peer of typescript@>=4.0 <4.2 but none is installed. You must install peer dependencies yourself.
npm WARN ng-packagr@11.2.4 requires a peer of typescript@>=4.0 <4.2 but none is installed. You must install peer dependencies yourself.
npm WARN @ngtools/webpack@11.2.14 requires a peer of typescript@~4.0.0 || ~4.1.0 but none is installed. You must install peer dependencies yourself.
npm WARN ajv-keywords@3.5.2 requires a peer of ajv@^6.9.1 but none is installed. You must install peer dependencies yourself.
npm WARN test_npm_publish@0.0.0 No repository field.\n\n
```

I found several existing discussions (some started in 2017) related to typescript and/or peer dependencies and am a bit confused on how to properly handle this:
- what is the recommended way to handle this? (this sounds like something everyone with angular would run into at some point)
  - will an explicit `"versionStrategy": "widen"` on typescript work (mentioned in https://github.com/renovatebot/renovate/issues/323#issuecomment-360153687)
  - I see also solutions with a dependencyDashboardApproval, (how) will that actually work with azure? it requires checkboxes in a dashboard issue? https://github.com/renovatebot/renovate/discussions/10030#discussioncomment-751501
- are there any plans for adjusting the package version update based on (unmet) peer dependency information (from npm)?

**Comment by viceice** [comment]
I'm using a custom preset for this

https://github.com/visualon/renovate-config/blob/main/ng.json
